lib/rubocop/cop/chef/modernize/cron_d_file_or_template.rb

```diff
@@ -81,7 +81,12 @@ class CronDFileOrTemplate < Base
 
           def on_block(node)
             file_or_template?(node) do |file_name|
+              # weed out types we can't evaluate
+              return unless file_name.dstr_type? || file_name.str_type?
+
+              # weed out string values that aren't a cron.d file
               return unless file_name.value.start_with?('/etc/cron.d/')
+
```

Comment:
I have the spec's passing with this diff if you would prefer it over checking the file types in the matcher macro:

```
diff --git a/lib/rubocop/cop/chef/modernize/cron_d_file_or_template.rb b/lib/rubocop/cop/chef/modernize/cron_d_file_or_template.rb
index c14523bd..99eea65f 100644
--- a/lib/rubocop/cop/chef/modernize/cron_d_file_or_template.rb
+++ b/lib/rubocop/cop/chef/modernize/cron_d_file_or_template.rb
@@ -48,6 +48,10 @@ module RuboCop
         #     action :delete
         #   end
         #
+        #   file "/etc/cron.d/#{job_name}" do
+        #     action :delete
+        #   end
+        #
         #   #### correct
         #   cron_d 'backup' do
         #     minute '1'
@@ -70,7 +74,7 @@ module RuboCop
           def_node_matcher :file_or_template?, <<-PATTERN
             (block
               (send nil? {:template :file :cookbook_file}
-                (str $_))
+                {(str $_) | (dstr (str $_) ...)})
               ...
             )
           PATTERN
```

Reply:
Sorry I wasn't sure if you were around. If you want to switch it to the node syntax that'd be great in another PR